pkg/apis/gcp/storage/v1alpha1/types.go

```diff
@@ -0,0 +1,936 @@
+}
+
+// IsAvailable for usage/binding
+func (b *Bucket) IsAvailable() bool {
+	return b.Status.IsReady()
+}
+
+// IsBound determines if the resource is in a bound binding state
+func (b *Bucket) IsBound() bool {
+	return b.Status.Phase == corev1alpha1.BindingStateBound
+}
+
+// SetBound sets the binding state of this resource
+func (b *Bucket) SetBound(state bool) {
+	if state {
+		b.Status.Phase = corev1alpha1.BindingStateBound
+	} else {
+		b.Status.Phase = corev1alpha1.BindingStateUnbound
+	}
+}
+
+// NewBucketSpec constructs Spec for this resource from the properties map
+func NewBucketSpec(p map[string]string) *BucketSpec {
+	var encryption *BucketEncryption
+	if v, found := p["encryptionDefaultKmsKeyName"]; found {
+		encryption = &BucketEncryption{DefaultKMSKeyName: v}
+	}
+
+	lifecycle := &Lifecycle{}
+	if v, found := p["lifecycle"]; found {
+		lifecycle = parseLifecycle(v)
+	}
+
+	var labels map[string]string
+	if v, found := p["labels"]; found {
+		labels = parseMap(v)
+	}
+
+	var logging *BucketLogging
+	if v, found := p["logging"]; found {
+		logging = parseLogging(v)
+	}
+
+	var website *BucketWebsite
+	if v, found := p["website"]; found {
+		website = parseWebsite(v)
+	}
+
+	bua := BucketUpdatableAttrs{
+		CORS:                       parseCORSList(p["cors"]),
+		DefaultEventBasedHold:      parseBool(p["defaultEventBasedHold"]),
+		Encryption:                 encryption,
+		Labels:                     labels,
+		Lifecycle:                  *lifecycle,
+		Logging:                    logging,
+		Website:                    website,
+		PredefinedACL:              p["predefinedACL"],
+		PredefinedDefaultObjectACL: p["predefinedDefaultObjectACL"],
+	}
+
+	bsa := BucketSpecAttrs{
+		BucketUpdatableAttrs: bua,
+		ACL:                  parseACLRules(p["acl"]),
+		DefaultObjectACL:     parseACLRules(p["defaultObjectACL"]),
+		Location:             p["location"],
+		StorageClass:         p["storageClass"],
+	}
+
+	return &BucketSpec{
+		BucketSpecAttrs: bsa,
+		ReclaimPolicy:   corev1alpha1.ReclaimRetain,
+	}
+}
+
+// parseBool returns true IFF string value is "true" or "True"
+func parseBool(s string) bool {
+	b, err := strconv.ParseBool(s)
+	if err != nil {
+		return false
+	}
+	return b
+}
+
+// parseMap string encoded map values
+// example: "foo:bar,one:two" -> map[string]string{"foo":"bar","one":"two"}
+func parseMap(s string) map[string]string {
+	m := map[string]string{}
+	for _, cfg := range strings.Split(s, ",") {
+		if kv := strings.SplitN(cfg, ":", 2); len(kv) == 2 {
+			m[strings.TrimSpace(kv[0])] = strings.TrimSpace(kv[1])
+		}
+	}
+	return m
+}
+
+// parseCORSList from json encoded string, return nil on error (if any)
+func parseCORSList(s string) []CORS {
+	var list []CORS
+	_ = json.Unmarshal([]byte(s), &list) // nolint: gosec
+	return list
+}
+
+// parseLifecycle parses json encoded lifecycle string
+func parseLifecycle(s string) *Lifecycle {
+	l := &Lifecycle{}
+	_ = json.Unmarshal([]byte(s), l) // nolint: gosec
```

Comment:
Ah, yes, this was my initial attempts battling linter warnings, before I found `//nolint` annotation. Removing.

Reply:
And... putting it back. So the deal is: `_ = json.Unmarshal(...)` considers error is being "handled" (by being ignored). If I switch to simply `json.Unmarshal(...)`, then I need add another `// nolint:  errcheck` annotation to avoid:
```
pkg/apis/gcp/storage/v1alpha1/types.go:934:16: Error return value of `json.Unmarshal` is not checked (errcheck)
        json.Unmarshal([]byte(s), &rules) // nolint: gosec
```